src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -161,12 +161,14 @@ const useCloudEnvironmentPolling = namespace => {
   }
   const load = async () => {
     try {
-      const [newDisks, newClusters] = await Promise.all([
+      const [newDisks, newClusters, galaxyDisks] = await Promise.all([
         Ajax(signal).Disks.list({ googleProject: namespace, creator: getUser().email }),
-        Ajax(signal).Clusters.list({ googleProject: namespace, creator: getUser().email })
+        Ajax(signal).Clusters.list({ googleProject: namespace, creator: getUser().email }),
+        Ajax(signal).Disks.list({ googleProject: namespace, creator: getUser().email, saturnApplication: 'galaxy' })
       ])
+      const galaxyDiskNames = _.map(disk => disk.name, galaxyDisks)
       setClusters(newClusters)
-      setPersistentDisks(newDisks)
+      setPersistentDisks(_.remove(disk => _.includes(disk.name, galaxyDiskNames), newDisks))
```

Comment:
I haven't tested yet, but I think you need `_.filter` instead of `_.remove`. `_.remove` mutates the array it's given and returns the removed elements.

Reply:
You are right, and I am not sure why it was working… /shrug

Reply:
Well... apparently it was working because... I guess lodash/fp changes the behavior of `_.remove`??? I compared the REPL in the lodash docs and the console of Terra UI and they do indeed behave differently. lodash/fp's `_.remove` apparently returns the new array with the elements removed and does not modify the given array... and the FP Guide does not make any mention of this that I can tell! Not gonna lie, I'm kinda mad right now...

Anyway, I still prefer using `_.filter` because it behaves _predictably_ and _as documented_ in lodash/fp.

Reply:
Coming back to this, I think I'm changing my mind. Despite the fact that the lodash FP Guide says nothing about this other than "immutable auto-curried iteratee-first data-last methods", this does seem to be an intentional behavior change in lodash/fp. Furthermore, we are using `_.remove` in several other places already. It's also more clear than having to negate the condition given to `_.filter`, so I think it's better to keep it as `_.remove`.

I regret the time and energy taken by this thread. I hope that it ultimately pays off in learning (though I have a feeling I've been down this road before myself 😞 ).

Reply:
Absolutely. And yeah, it would be very nice if they had documented the changes at all. /ugh 

I will revert it.